rust/numaflow-core/src/metrics.rs

```diff
@@ -793,6 +847,89 @@ impl Drop for PendingReaderTasks {
     }
 }
 
+// TODO(lookback) - using new implementation for monovertex right now,
+// deprecate old implementation and use this for pipeline as well once
+// corresponding changes are completed.
+/// When the PendingReaderTasks_ is dropped, we need to clean up the pending exposer and the pending builder tasks.
+impl Drop for PendingReaderTasks_ {
+    fn drop(&mut self) {
+        self.expose_handle.abort();
+        info!("Stopped the Lag-Reader Expose tasks");
+    }
+}
+
+// TODO(lookback) - using new implementation for monovertex right now,
+// deprecate old implementation and use this for pipeline as well once
+// corresponding changes are completed.
+
+// Periodically exposes the pending metrics by calculating the average pending messages over different intervals.
+async fn expose_pending_metrics_(
+    mut lag_reader: LagReader,
+    lag_checking_interval: Duration,
+    is_mono_vertex: bool,
+) {
+    let mut ticker = time::interval(lag_checking_interval);
+
+    loop {
+        ticker.tick().await;
+
+        match &mut lag_reader {
+            LagReader::Source(source) => match fetch_source_pending(source).await {
+                Ok(pending) => {
+                    if pending != -1 {
+                        info!("Pending messages {:?}", pending);
+                        if is_mono_vertex {
+                            let metric_labels = mvtx_forward_metric_labels().clone();
```

Comment:
Try to see if we can avoid level 2 branching.

Reply:
Can do refactor in a separate PR if you're fine with this right now!